docs/import_specifications.ADR.md

```diff
@@ -0,0 +1,186 @@
+# Import Specifications Architecture Design Record
+
+This document specifies the design for handling import specifications in the Staging Service (StS).
+An upload specification is an Excel, CSV, or TSV file that contains instructions for how
+to import one or more files in the staging area to KBase as KBase data types.
+
+## Resources
+
+* [The original strategy document for this approach](https://docs.google.com/document/d/1ocmZVBlTzAh_cdZaWGRwIbAuH-mcPRZFdhcwRhAfzxM/edit)
+  * Readable for everyone that has access to the KBase Google Docs folder
+  * The document contains short descriptions of future projects not included in this work,
+    such as generating template files on the fly and rich templates.
+
+## Front end changes
+
+The design introduces a new StS data type, `upload_specification`. The FE's current 
+behavior is to display any data types returned from the StS in the file dropdown, but silently
+ignore user-selected files for which the selected data type is unknown to the narrative, a bug.
+The FE will be updated to ignore unknown data types returned from the StS, allowing for phased,
+non-lockstep upgrades. This work is not included in this project, but will be in a future FE
+project.
+
+## Upload specification input files
+
+Input file formats may be Excel, CSV, or TSV. An example CSV file structure for GFF/FASTA uploads
+is below:
+
+```
+Data type:, gff_metagenome
+fasta_file, gff_file, genome_name, source, release, genetic_code, generate_missing_genes
+FASTA File Path, GFF3 File Path, Metagenome Object Name, Source of metagenome, Release or Version of the Source Data, Genetic Code for protein translation, Spoof Genes for parentless CDS
+mygenome.fa, mygenome.gff3, mygenomeobject, , , 11, 0
+mygenome2.fa, mygenome2.gff3, mygenomeobject2, yermumspoo, 30456, 11, 1
+...
+```
+
+The file, by row, is:
+1. The data type, in this case `gff_metagenome`. This is a data type listed in the
+  [Mappings.py](https://github.com/kbase/staging_service/blob/master/staging_service/autodetect/Mappings.py)
+  file in the StS. The Narrative is expected to understand these types and map them to uploader
+  apps.
+2. The IDs of the app inputs from the `spec.json` file. 
+3. The corresponding human readable names of the app inputs from the `display.yaml` file.
+4. (and beyond) Import specifications. Each line corresponds to a single import.
+
+For Excel files, the first two rows may be hidden in any provided templates. Additionally,
+Excel files may contain multiple data types, one per tab. Empty tabs will be ignored, and tabs
```

Comment:
Additional question after view the template: does "empty" here mean truly empty or only contains the pre-filled info from the template? Or TBD? We'll need to clarify in the docs.

Reply:
Truly empty, although if is just has the pre-filled info that tab will be effectively ignored as well since there's not data.

So I guess both of your definitions apply.

Reply:
So if they download the template and don't fill it a tab but don't delete it either, that would work? That would be ideal as it would save them one small step.

Reply:
yep, it'd just be ignored